### PR TITLE
When viewing a file, hide the add button

### LINKED ISSUE #25281: Should check where the tree path is a directory or a file when create a new file in web ui
### Description

Steps:
- click a file in your repo
- click `Add File` -> `New File`

image

### Gitea Version

latest

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

build

### Database

None

### PR DESCRIPTION
**Pull request by hiifong**
Fix #25281
When viewing a file, hide the add button
image  
image

### CODE REVIEW

File: templates/repo/home.tmpl [hunk 1]
```diff
@@ -84,23 +84,25 @@
 				{{end}}
 
 				{{if and .CanWriteCode .IsViewBranch (not .Repository.IsArchived)}}
```
Comment (silverwind) [inline templates/repo/home.tmpl]:
```suggestion
				{{if and .CanWriteCode .IsViewBranch (not .Repository.IsArchived) (not .IsViewFile)}}
```

Same effect but avoid having to re-indent.

**Review by silverwind — APPROVED** [review]

**Comment by hiifong** [comment]
```diff
diff --git a/templates/repo/home.tmpl b/templates/repo/home.tmpl
index 8589574a5..616874d26 100644
--- a/templates/repo/home.tmpl
+++ b/templates/repo/home.tmpl
@@ -83,18 +83,29 @@
 					<a href="{{.Repository.Link}}/find/{{.BranchNameSubURL}}" class="ui compact basic button">{{.locale.Tr "repo.find_file.go_to_file"}}</a>
 				{{end}}
 
-				{{if and .CanWriteCode .IsViewBranch (not .Repository.IsArchived) (not .IsViewFile)}}
+				{{if and .CanWriteCode .IsViewBranch (not .Repository.IsArchived)}}
 					<button class="ui dropdown basic compact jump button gt-mr-2"{{if not .Repository.CanEnableEditor}} disabled{{end}}>
 						{{.locale.Tr "repo.editor.add_file"}}
 						{{svg "octicon-triangle-down" 14 "dropdown icon"}}
 						<div class="menu">
-							<a class="item" href="{{.RepoLink}}/_new/{{.BranchName | PathEscapeSegments}}/{{.TreePath | PathEscapeSegments}}">
-								{{.locale.Tr "repo.editor.new_file"}}
-							</a>
-							{{if .RepositoryUploadEnabled}}
-							<a class="item" href="{{.RepoLink}}/_upload/{{.BranchName | PathEscapeSegments}}/{{.TreePath | PathEscapeSegments}}">
-								{{.locale.Tr "repo.editor.upload_file"}}
-							</a>
+							{{if .IsViewFile}}
+								<a class="item" href="{{.RepoLink}}/_new/{{.BranchName | PathEscapeSegments}}{{.ParentPath}}">
+									{{.locale.Tr "repo.editor.new_file"}}
+								</a>
+								{{if .RepositoryUploadEnabled}}
+								<a class="item" href="{{.RepoLink}}/_upload/{{.BranchName | PathEscapeSegments}}{{.ParentPath}}">
+									{{.locale.Tr "repo.editor.upload_file"}}
+								</a>
+								{{end}}
+							{{else}}
+								<a class="item" href="{{.RepoLink}}/_new/{{.BranchName | PathEscapeSegments}}/{{.TreePath | PathEscapeSegments}}">
+									{{.locale.Tr "repo.editor.new_file"}}
+								</a>
+								{{if .RepositoryUploadEnabled}}
+								<a class="item" href="{{.RepoLink}}/_upload/{{.BranchName | PathEscapeSegments}}/{{.TreePath | PathEscapeSegments}}">
+									{{.locale.Tr "repo.editor.upload_file"}}
+								</a>
+								{{end}}
 							{{end}}
 							<a class="item" href="{{.RepoLink}}/_diffpatch/{{.BranchName | PathEscapeSegments}}/{{.TreePath | PathEscapeSegments}}">
 								{{.locale.Tr "repo.editor.patch"}}
```

**Comment by hiifong** [comment]
Is this right?

**Comment by silverwind** [comment]
> Keep the add file button shown, but use its parent path as the new file's path if the current file is being viewed

Hmm, I'm not sure if this is really needed, it seems a bit unexpected behaviour imho.

**Review by lunny — APPROVED** [review]

**Comment by silverwind** [comment]
Let's merge this as-is. Simple and does what it says.

**Comment by hiifong** [comment]
> I'd say we merge this as-is. Simple and does what it says. @hiifong ok with you?

OK

**Pull request merged**